src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java

```diff
@@ -1,34 +1,52 @@
 package edu.kis.vh.nursery;
 
 public class DefaultCountingOutRhymer {
+    private static final int DEFAULT_RETURN_VALUE = -1;
+    private static final int MAX_STACK_SIZE = 12;
+    private int[] numbers = new int[getMaxStackSize()];
 
-    private int[] numbers = new int[12];
+    private int total = getDefaultReturnValue();
 
-    public int total = -1;
+    public static int getDefaultReturnValue() {
+        return DEFAULT_RETURN_VALUE;
+    }
+
+    public static int getMaxStackSize() {
+        return MAX_STACK_SIZE;
+    }
 
-    public void countIn(int in) {
+    protected void countIn(int in) {
         if (!isFull())
-            numbers[++total] = in;
+            getNumbers()[++total] = in;
     }
 
-    public boolean callCheck() {
-        return total == -1;
+    protected boolean callCheck() {
+        return total == getDefaultReturnValue();
     }
 
-    public boolean isFull() {
-        return total == 11;
+    protected boolean isFull() {
+        return total == getMaxStackSize() -1;
     }
 
-    protected int peekaboo() {
+    public int peekaboo() {
         if (callCheck())
-            return -1;
-        return numbers[total];
+            return getDefaultReturnValue();
+        return getNumbers()[total];
     }
 
     public int countOut() {
         if (callCheck())
-            return -1;
-        return numbers[total--];
+            return getDefaultReturnValue();
+        return getNumbers()[total--];
+    }
+
+    public int getTotal() {
+        return total;
+    }
+
+
+    public int[] getNumbers() {

```

Comment:
Shouldn't be encapsulated (typical private field). This is an exaggeration. Using private fields internally usually do not require encapsulation.